pubtools/pulplib/_impl/model/repository/yum.py

```diff
@@ -90,3 +90,22 @@ class YumRepository(Repository):
         type=list,
         converter=FrozenList,
     )
+
+    arch = pulp_attrib(default=None, type=str, pulp_field="notes.arch")
+    """Repository note describing architecture of the repository"""
+
+    platform_full_version = pulp_attrib(
```

Comment:
I'm not entirely confident about adding these two platform version fields here.

If I remember correctly - these two are quite messy in cdn-utils and it's not really clear what they're supposed to mean.  Docs here don't really help, they just repeat the attribute name without explaining what it means.

Part of the reason to go through this process of modelling all the data into classes is to get away from this legacy with so many poorly defined fields which could contain anything, nobody really knows what they're supposed to be and the only way to figure it out is to read all the code using them.  If we just expose those legacy fields under the same name again without any kind of validation, conversion or explanations for their meaning then we're not improving things.

I really don't want to accept these unless:

- we can provide a proper explanation for what the values are (e.g. not just say it's a full version but explain what exactly that is, x.y.z ?  And explain what is 'platform')
- we can do something to clean up nonsensical values.  e.g. in cdn-utils right now I'm seeing values of '5.9' and $releasever for platform_minor_version.  These seem to be wrong because they aren't minor versions (e.g. in 5.9.0, the minor version would be 9).  I don't want to just "pass on" whatever wrongness already exists.

Can we make it better somehow?  Also if you point at ubipop change which wants to use the fields, I might be able to make some suggestion.

Reply:
Yeah, those fields are not really clear, actually I just want to refactor ubipop a bit and use pulplib Client instead of the custom one in ubipop because the changes for improving performance will be easier when it uses pulplib Client.

In ubipop there are use-cases for those repo notes:
1) Find the counterpart repo of given ubi repo
https://github.com/release-engineering/ubi-population-tool/blob/920b2890aa95a9701288fea2388dd7ea196abefb/ubipop/__init__.py#L325
This can be most probably dropped as we already have 'population_sources' repo note that is calculated in cdn-utils for all ubi repos. 

2) It's used for getting proper branch of ubi-config:
https://github.com/release-engineering/ubi-population-tool/blob/920b2890aa95a9701288fea2388dd7ea196abefb/ubipop/__init__.py#L215 

Basically it needs to know whether the repo is mainline or not and take the proper version of ubi-config either based on ubi_config_version (for mainline repo) or based on platform_full_version for DOT repos. If we could stop populating DOT repos, it would be a way easier. But ideally we should populate 'ubi_config_version' also for DOT repos then we don't need to introduce those repo notes in pulplib.

Reply:
OK, for (1) I would definitely be in favor of just setting ubi_config_version across *all* repos rather than leaving it blank sometimes and requiring calculation of a default value based on other notes.  If that wouldn't work for some reason, I'd actually prefer to move the calculation of default ubi_config_version into this project rather than have it in ubipop.

For (2) I didn't quite understand it yet, but from your comment it sounds like you're saying that if we populated ubi_config_version everywhere, it would also solve that case.  If that's the case then that sounds like an ideal path forward to me.